### PR TITLE
CPP-747 Fix case sensitive keyspaces

### PR DESCRIPTION
**Pull request by mpenick**

### CODE REVIEW

File: src/hash_table.hpp [hunk 1]
```diff
@@ -72,7 +72,6 @@ class CaseInsensitiveHashTable : public Allocated {
 
 private:
   size_t index_mask_;
-  size_t count_;
```
Comment (mpenick) [inline src/hash_table.hpp]: Unused variable. Unrelated.

File: src/utils.cpp [hunk 2]
```diff
@@ -111,37 +111,21 @@ String& trim(String& str) {
...
+  for (String::const_iterator it = str.begin() + 1, end = str.end(); it != end; ++it) {
+    char c = *it;
+    if (!((c >= '0' && c <= '9') || (c == '_') || (c >= 'a' && c <= 'z'))) {
       return false;
     }
   }
   return true;
 }
 
-bool is_valid_lower_cql_id(const String& str) {
```
Comment (mpenick) [inline src/utils.cpp]: This was implemented incorrectly.

File: src/utils.cpp [hunk 3]
```diff
@@ -159,18 +143,7 @@ String& quote_id(String& str) {
   return str;
 }
 
-String& escape_id(String& str) { return is_valid_lower_cql_id(str) ? str : quote_id(str); }
-
-String& to_cql_id(String& str) {
```
Comment (mpenick) [inline src/utils.cpp]: `to_cql_id()` was dead code.

File: tests/src/integration/tests/test_use_keyspace.cpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+class UseKeyspaceCaseSensitiveTests : public Integration {
+public:
+  UseKeyspaceCaseSensitiveTests() {}
+
+  // Make a case-sensitive keyspace capitalizing the first char and wrapping in double quotes
+  virtual std::string default_keyspace() {
+    std::string temp(Integration::default_keyspace());
+    temp[0] = std::toupper(temp[0]);
+    return "\"" + temp + "\"";
+  }
```
Comment (mikefero) [inline tests/src/integration/tests/test_use_keyspace.cpp]: Smart!

**Pull request merged**